generator/views/funnel_analysis_view.py

```diff
@@ -0,0 +1,238 @@
+        return len([k for k in self.tables[0] if k.startswith("event_type_")])
+
+    def _funnel_analysis_lookml(self) -> List[Dict[str, Any]]:
+        dimensions = [
+            {
+                "name": f"completed_event_{n}",
+                "type": "yesno",
+                "description": f"Whether the user completed step {n} on the associated day.",
+                "sql": (
+                    "REGEXP_CONTAINS(${TABLE}.events, mozfun.event_analysis.create_funnel_regex(["
+                    f"${{event_type_{n}.match_string}}],"
+                    "True))"
+                ),
+            }
+            for n in range(1, self.n_events() + 1)
+        ]
+        count_measures: List[Dict[str, Any]] = [
+            {
+                "name": f"count_user_days_event_{n}",
+                "description": (
+                    f"The number of user-days that completed step {n}. "
+                    "Grouping by day makes this is a count of users."
+                ),
+                "type": "count",
+                "filters": [{f"completed_event_{ni}": "yes"} for ni in range(1, n + 1)],
+            }
+            for n in range(1, self.n_events() + 1)
+        ]
+        fractional_measures: List[Dict[str, Any]] = [
+            {
+                "name": f"fraction_user_days_event_{n}",
+                "description": f"Of the user-days that completed Step 1, the fraction that completed step {n}.",
+                "type": "number",
+                "sql": f"SAFE_DIVIDE(${{count_user_days_event_{n}}}, ${{count_user_days_event_1}})",
+            }
+            for n in range(1, self.n_events() + 1)
+        ]
+        return [
+            {
+                "name": "funnel_analysis",
+                "extends": ["events_daily_table"],
+                "dimensions": dimensions,
+                "measures": count_measures + fractional_measures,
+            }
+        ]
+
+    def _event_types_lookml(self) -> List[Dict[str, Any]]:
+        events = (
+            [
+                {
+                    "name": "event_types",
+                    "derived_table": {
+                        "sql": (
+                            "SELECT "
+                            "mozfun.event_analysis.aggregate_match_strings( "
+                            "ARRAY_AGG( "
+                            "mozfun.event_analysis.event_index_to_match_string(index))) AS match_string "
+                            "FROM "
+                            f"{self.tables[0]['event_types']} "
+                            "WHERE "
+                            "{% condition category %} category {% endcondition %} "
+                            "AND {% condition event %} event {% endcondition %}"
+                        )
+                    },
+                    "filters": [
+                        {
+                            "name": "category",
+                            "description": "The event category, as defined in metrics.yaml.",
```

Comment:
Let's go for something more human centred here. I'm assuming you mean metric category as defined inside Glean?

```suggestion
                            "description": "The Glean metric category, as defined inside the application.",
```